src/main/java/com/auctionapp/api/controller/CategoryController.java

```diff
@@ -0,0 +1,37 @@
+package com.auctionapp.api.controller;
+
+import java.util.List;
+
+import com.auctionapp.api.model.dto.CategoryDto;
+import com.auctionapp.api.service.CategoryService;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/categories")
+public class CategoryController {
+	
+	private final CategoryService service;
+
+	public CategoryController(final CategoryService categoryService) {
+		this.service = categoryService;
+	}
+
+	@GetMapping
+	public ResponseEntity<List<CategoryDto>> getLandingPageCategories() {
```

Comment:
The backend shouldn't know that there is a landing page it's providing info for, if you want to fetch top 9 categories, just call the method that. Also I believe the 9 categories in the design are just an example and you should always fetch and display all of them.

Reply:
I have renamed the method.

I am not sure that all categories should be fetched for the landing page, because these is the last row in the landing page list of categories that says 'All categories' and leads to shop page with all categories listed, so that is why I put only 9 categories on the landing page. Also, I think it would mess up the design of the page if there are 20 categories displayed for example. Is it okay if I leave it like this?

Reply:
Yes, in that case, you should definitely leave it as is, because even disregarding the frontend, it's a better way because there could (not likely, but potentially) be thousands of categories which we'd then want to paginate in some way and not just fetch all at once. I was looking at old designs again, my bad.